client/gutenberg/extensions/map/edit.js

```diff
@@ -89,23 +86,28 @@ class MapEdit extends Component {
 	};
 	apiCall( serviceApiKey = null, method = 'GET' ) {
 		const { noticeOperations } = this.props;
+		const { apiKey } = this.state;
+		this.setState( { apiRequestOutstanding: true } );
 		const url = '/?rest_route=/jetpack/v4/service-api-keys/mapbox';
 		const fetch = serviceApiKey
 			? { url, method, data: { service_api_key: serviceApiKey } }
 			: { url, method };
+		this.setState( { apiRequestOutstanding: true } );
 		apiFetch( fetch ).then(
 			result => {
 				noticeOperations.removeAllNotices();
 				this.setState( {
 					apiState: result.service_api_key ? API_STATE_SUCCESS : API_STATE_FAILURE,
 					apiKey: result.service_api_key,
 					apiKeyControl: result.service_api_key,
+					apiRequestOutstanding: false,
 				} );
 			},
 			result => {
 				this.onError( null, result.message );
 				this.setState( {
-					apiState: API_STATE_FAILURE,
+					apiRequestOutstanding: false,
```

Comment:
We might never hit this, let's set to false in a `.finally()`.

Simple payments has the flow of with `setState` callback and finally I describe:

https://github.com/Automattic/wp-calypso/blob/5f0fa6fe68943b2bfb4aea83592aea04ee36a07a/client/gutenberg/extensions/simple-payments/edit.js#L111-L112

https://github.com/Automattic/wp-calypso/blob/5f0fa6fe68943b2bfb4aea83592aea04ee36a07a/client/gutenberg/extensions/simple-payments/edit.js#L134-L138

Reply:
I have a couple of questions about this one - going to leave this out for the moment.

Reply:
> I have a couple of questions about this one - going to leave this out for the moment.

Correction:

I'd misread this and see this uses `.then( onFulfilled, onRejected );` which sets `apiRequestOutstanding: false` in both cases 👍